src/dev.rs

```diff
@@ -407,24 +410,43 @@ impl<F: FieldExt> MockProver<F> {
 
         let mut cs = ConstraintSystem::default();
         let config = ConcreteCircuit::configure(&mut cs);
+        let cs = cs;
 
-        let fixed = vec![vec![None; n as usize]; cs.num_fixed_columns];
-        let advice = vec![vec![None; n as usize]; cs.num_advice_columns];
-        let permutations = cs
-            .permutations
-            .iter()
-            .map(|p| permutation::keygen::Assembly::new(n as usize, p))
-            .collect();
+        if n < cs.minimum_rows() {
+            return Err(Error::NotEnoughRowsAvailable);
+        }
 
+        if instance.len() != cs.num_instance_columns {
+            return Err(Error::IncompatibleParams);
+        }
+
+        let instance = instance
+            .into_iter()
+            .map(|mut instance| {
+                if instance.len() > n - (cs.blinding_factors() + 1) {
```

Comment:
```suggestion
        let blinding_factors = cs.blinding_factors();
        let usable_rows = n - (blinding_factors + 1);
        let instance = instance
            .into_iter()
            .map(|mut instance| {
                if instance.len() > usable_rows {
```

Reply:
I don't see why we're mentioning this here, this is just dev tooling?

Reply:
Ah, I was just responding to @str4d's comment asking for more documentation. There's a case for adding `cs.usable_rows()`, but we shouldn't block the PR on that.